File: sdk/php/src/Resources/Support.php
<?php

namespace HiTechCloud\SDK\Resources;

class Support extends BaseResource
{
    /**
     * List tickets
     */
    public function listTickets(array $params = []): array
    {
        return $this->http->get('/api/tickets', $params);
    }

    /**
     * Create ticket
     */
    public function createTicket(array $data): array
    {
        return $this->http->post('/api/tickets', $data);
    }

    /**
     * Get ticket
     */
    public function getTicket(string $ticketNumber): array
    {
        return $this->http->get('/api/tickets/' . $ticketNumber);
    }

    /**
     * Reply to ticket
     */
    public function replyTicket(string $ticketNumber, array $data): array
    {
        return $this->http->post('/api/tickets/' . $ticketNumber, $data);
    }

    /**
     * Get attachment
     */
    public function getAttachment(string $file): array
    {
        return $this->http->get('/api/ticket/attachment/' . $file);
    }

    /**
     * Open ticket
     */
    public function openTicket(string $ticketNumber): array
    {
        return $this->http->put('/api/tickets/' . $ticketNumber . '/open');
    }

    /**
     * Close ticket
     */
    public function closeTicket(string $ticketNumber): array
    {
        return $this->http->put('/api/tickets/' . $ticketNumber . '/close');
    }

    /**
     * Get departments
     */
    public function getDepartments(): array
    {
        return $this->http->get('/api/ticket/departments');
    }

    /**
     * List news
     */
    public function listNews(array $params = []): array
    {
        return $this->http->get('/api/news', $params);
    }

    /**
     * Get news article
     */
    public function getNews(int $newsId): array
    {
        return $this->http->get('/api/news/' . $newsId);
    }

    /**
     * List knowledgebase
     */
    public function listKnowledgebase(array $params = []): array
    {
        return $this->http->get('/api/knowledgebase', $params);
    }

    /**
     * Get KB category
     */
    public function getKnowledgebaseCategory(int $categoryId): array
    {
        return $this->http->get('/api/knowledgebase/' . $categoryId);
    }

    /**
     * Get KB article
     */
    public function getKnowledgebaseArticle(int $articleId): array
    {
        return $this->http->get('/api/knowledgebase/article/' . $articleId);
    }

}

File: sdk/csharp/Resources/Support.cs
namespace HiTechCloud.SDK.Resources;

public class Support : BaseResource
{
    public Support(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> TicketsAsync()
    {
        return await Http.GetAsync("/api/tickets");
    }

    public async Task<Dictionary<string, object?>> CreateTicketAsync(Dictionary<string, object> data)
    {
        return await Http.PostAsync("/api/tickets", data);
    }

    public async Task<Dictionary<string, object?>> TicketAsync(string number)
    {
        return await Http.GetAsync(string.Format("/api/tickets/{number}", number));
    }

    public async Task<Dictionary<string, object?>> ReplyTicketAsync(string number, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/tickets/{number}", number), data);
    }

    public async Task<Dictionary<string, object?>> TicketAttachmentAsync(string file)
    {
        return await Http.GetAsync(string.Format("/api/ticket/attachment/{file}", file));
    }

    public async Task<Dictionary<string, object?>> OpenTicketAsync(string number)
    {
        return await Http.PutAsync(string.Format("/api/tickets/{number}/open", number), null);
    }

    public async Task<Dictionary<string, object?>> CloseTicketAsync(string number)
    {
        return await Http.PutAsync(string.Format("/api/tickets/{number}/close", number), null);
    }

    public async Task<Dictionary<string, object?>> DepartmentsAsync()
    {
        return await Http.GetAsync("/api/ticket/departments");
    }

    public async Task<Dictionary<string, object?>> NewsAsync()
    {
        return await Http.GetAsync("/api/news");
    }

    public async Task<Dictionary<string, object?>> NewsItemAsync(int newsId)
    {
        return await Http.GetAsync(string.Format("/api/news/{newsId}", newsId));
    }

    public async Task<Dictionary<string, object?>> KnowledgebaseAsync()
    {
        return await Http.GetAsync("/api/knowledgebase");
    }

    public async Task<Dictionary<string, object?>> KnowledgebaseCategoryAsync(int categoryId)
    {
        return await Http.GetAsync(string.Format("/api/knowledgebase/{categoryId}", categoryId));
    }

    public async Task<Dictionary<string, object?>> KnowledgebaseArticleAsync(int articleId)
    {
        return await Http.GetAsync(string.Format("/api/knowledgebase/article/{articleId}", articleId));
    }
}
